### DISCUSSION TITLE
Second bar on the right side

### DISCUSSION THREAD

**Discussion by F0rce** [coder/code-server#4586] [Ideas] [answered]
Is there a way in VSCode / code-server to add a bar on the right side, to allign some extensions / views on the other side ?

When having to use a lot of extensions, the left side is some kind of overflown. You have to continuously scroll up and down.

If there is no such way to do so, I think it could be a very helping addition to code-server and improve multitasking a lot, for example when you have to use two extensions at the same time.

I added a screenshot of what that could look like (Theia Blueprint).

![MicrosoftTeams-image (3)](https://user-images.githubusercontent.com/60381251/145012747-c47aa631-bd63-42e2-be54-b8d247b0ad5e.png)

**Comment by lg-nineoneninethreenine** [comment]
I don't see a way as of now, I kinda feel this is upstream, @jsjoeio any ideas on this?

**Comment by jsjoeio** [comment]
Hmm...this seems like something that would need to be implemented in VS Code. Does that type of functionality already exist?

**Reply by lg-nineoneninethreenine** [reply]
I don't think so...

**Comment by ghost** [comment, marked as answer]
https://github.com/microsoft/vscode/issues/132893

TLDR: its already in the recent stable versions. To enable add `"workbench.experimental.sidePanel.enabled": true,` in `settings.json`. Then reload (altho for me I don't have to) and then in the menubar `View -> Appearance -> Show Side Panel`

**Reply by jsjoeio** [reply]
TIL! Thanks for the links and explanation!

**Reply by F0rce** [reply]
Thank you very much :)

**Reply by F0rce** [reply]
@jsjoeio but is it already possible in v3.12.0 because I can't seem to get it running on code-server. But in my local installation of vscode, it seems to work. Should I just be hyped for v4.0.0 and try it then ?

**Reply by lg-nineoneninethreenine** [reply]
Maybe this is something I can answer. I think you should be hyped for 4.0.0 since I think it's not on the VS Code base of 3.12.0. But if you want to see if it's true, go ahead. I don't know if VS Code added it before code-server's 3.12.0 release or after.

**Reply by F0rce** [reply]
thanks for the answer :) I can say that with 3.12.0 it is not working. So I will wait and try it as soon as 4.0.0 releases. 

Have a nice weekend (forgot that it is friday),
David

**Reply by lg-nineoneninethreenine** [reply]
Thanks 😁

**Reply by jsjoeio** [reply]
> but is it already possible in v3.12.0 because I can't seem to get it running on code-server.

Looking at the upstream issue, I'm guessing this came in a latter VS Code version so it makes sense it's not available in v3.12.0.

Yes, you should be hyped for 4.0.0 - it _should_ be available there! We're going to do a pre-release today so keep your eyes peeled!
